src/timeline/js/controllers.js

```diff
@@ -715,11 +715,16 @@ App.controller("TimelineCtrl", function ($scope) {
       }
     }
     // Resize timeline
-    if (furthest_right_edge > $scope.project.duration) {
+    if (furthest_right_edge > $scope.project.duration || $scope.project.duration > furthest_right_edge + 20) {
```

Comment:
Let's move these constants (10, 20) into a variable for clarity

Reply:
Also, we might need to incorporate a "minimum" duration into this logic. Not sure though, but just a guess. For example, let's say we add a 60 min clip, and then delete it. Does the timeline get resized to 0? Does that have any effect on the size of the timeline as shown to the user?

Reply:
It would reset to 10, so that's a good point. I'll make sure the minimum is what the timeline starts off at for a new project.